sherpa/astro/io/__init__.py

```diff
@@ -1287,9 +1349,9 @@ def _reconstruct_rmf(rmf):
     """
 
     n_grp = []
```

Comment:
While you are at it, why not type `n_grp`? I don't think either is critical, but since we type the others it'll be nice to be consistent.

Reply:
I was trying to get by with the minimal amount of annotations to please `mypy` but I will note

- it can be advantageous to the reader to know what you expect in somewhat complex cases, as you mention
- tracking the minimal set of annotations is a fools game, as it depends on that version you are using, what software you are using, and what other changes have been made to the code base

Saying that, I'd rather not make too many other changes here as that has an impact on the downstream PRs (in this case #1921 which is being reworked to match these changes in #2085). Will need to look at this a bit more closely.

Reply:
So, in #1921 (although #2085 is an easier one to follow) I change this code quite significantly, so I want to look at this issue in that PR and not here.